lang/cs/Org.Apache.REEF.IO/PartitionedData/FileSystem/FileSystemPartitionInputDataSet.cs

```diff
@@ -116,22 +120,31 @@ IEnumerator IEnumerable.GetEnumerator()
             return _partitions.Values.GetEnumerator();
         }
 
-        private string FormId(ISet<string> filePaths)
+        /// <summary>
+        /// The id is derived form the a input file name. For whatever reason if it doesn't work out, default will be used.
+        /// </summary>
+        /// <param name="filePaths"></param>
+        /// <returns></returns>
+        private static string FormId(ISet<string> filePaths)
         {
             string id = "";
-            if (filePaths != null && filePaths.Count > 0)
+            try
             {
-                var path = filePaths.First();
-                var paths = path.Split(new string[] {StringSeparators}, StringSplitOptions.None);
-                if (paths.Length > 0)
+                if (filePaths != null && filePaths.Count > 0)
                 {
-                    FileInfo fInfo = new FileInfo(paths[0]);
-                    if (fInfo.Directory != null)
+                    var path = filePaths.First();
+                    var paths = path.Split(new string[] {"/", "//", "\\"}, StringSplitOptions.None);
+                    if (paths.Length > 0)
                     {
-                        id = fInfo.Directory.Name;
+                        id = paths[paths.Length - 1];
                     }
                 }
             }
+            catch (Exception e)
+            {
+                Logger.Log(Level.Warning, string.Format(CultureInfo.CurrentCulture, "The filePaths cannot be parsed for generating dataset id", e));
+            }
+
             return IdPrefix + id;
```

Comment:
I don't see any advantage of using string format in this case. None of the advantages of using string format fit this case. Do we have coding guideline saying using string format in all the places where sting + is used? I don't think so.